src/oidcc_jwt_util.erl

```diff
@@ -218,6 +231,107 @@ sign(Jwt, Jwk, [Algorithm | RestAlgorithms], JwsFields0) ->
         _ -> sign(Jwt, Jwk, RestAlgorithms, JwsFields0)
     end.
 
+%% @private
+-spec decrypt_if_needed(
+    Jwt :: binary(),
+    Jwk :: jose_jwk:key(),
+    SupportedAlgorithms :: [binary()] | undefined,
+    SupportedEncValues :: [binary()] | undefined
+) ->
+    {ok, binary()} | {error, no_supported_alg_or_key}.
+decrypt_if_needed(Jwt, Jwk, SupportedAlgorithms, SupportedEncValues) ->
+    maybe
+        {ok, _Jwe} ?= jwe_peek_protected(Jwt),
```

Comment:
Why are we peeking into the jwt? The result isn’t used anywhere.

Is the goal to just validate the jet?